ion/textreader.go

```diff
@@ -237,7 +237,11 @@ func (t *textReader) nextBeforeTypeAnnotations() (bool, error) {
 				if err := t.verifyUnquotedSymbol(val, "annotation"); err != nil {
 					return false, err
 				}
+			} else if tok == tokenSymbolOperator {
+				return false, &SyntaxError{
+					"annotations with operator " + val + " should be enclosed in quotes", t.tok.Pos() - 1}
```

Comment:
Quoting from [the `symbols` section of the spec](https://amzn.github.io/ion-docs/docs/spec.html#symbol):

> Within S-expressions, the rules for unquoted symbols include another set of tokens: operators. An operator is an unquoted sequence of one or more of the following nineteen ASCII characters: !#%&*+-./;<=>?@^`|~ 

AFAIK, the concept of an 'operator' only exists within the context of s-expressions, which makes this error message rather confusing. I would generalize it a bit.

```suggestion
			} else if tok == tokenSymbolOperator {
				return false, &SyntaxError{
					"Annotations that include a '" + val + "' must be enclosed in quotes.", t.tok.Pos() - 1}
```